harper-cli/src/main.rs

```diff
@@ -142,16 +157,16 @@ fn main() -> anyhow::Result<()> {
 fn load_file(file: &Path) -> anyhow::Result<(Document, String)> {
     let source = std::fs::read_to_string(file)?;
 
-    let mut parser: Box<dyn harper_core::parsers::Parser> =
-        if let Some("md") = file.extension().map(|v| v.to_str().unwrap()) {
-            Box::new(Markdown)
-        } else {
-            Box::new(
+    let parser: Box<dyn harper_core::parsers::Parser> =
+        match file.extension().map(|v| v.to_str().unwrap()) {
+            Some("md") => Box::new(Markdown),
+            Some("typ") => Box::new(Typst),
```

Comment:
With the exception of adding the file extension here, the changes in this file aren't really relevant to the PR. I like them. Would you mind putting them in a different PR?